src/models/Collaborator.ts

```diff
@@ -52,6 +56,93 @@ export class TemplateCollaborator extends Collaborator {
     return this.errors.length <= 0;
   }
 
+  // Save the current record
+  async create(context: MyContext): Promise<TemplateCollaborator> {
+    // First make sure the record is valid
+    if (await this.isValid()) {
+      const current = await TemplateCollaborator.findByTemplateIdAndEmail(
+        'TemplateCollaborator.create',
+        context,
+        this.templateId,
+        this.email,
+      );
+
+      if (current) {
+        this.errors.push('Collaborator has already been added');
+      } else {
+        // Verify that the template we want to attach the collaborator to exists!
+        const templateExists = await Template.exists(
```

Comment:
L74: Should "Template.exists()" be "TemplateCollaborator.exists()"?

Reply:
It could technically be either since they both inherit from the MySqlModel and that it where the `exists` function lives. Since we're checking to see if the Template we want to attach the collaborator to exists, I thought it was more readable to use Template here.

Reply:
Ok thanks @briri. If that's the case, you might want to change the "TemplateCollaborator.exists()" on line 115 to "Template.exists()" just for consistency.